docs/contributing/the_basics.md

```diff
@@ -7,65 +7,141 @@ An overview on contributing to the _Black_ project.
 Development on the latest version of Python is preferred. As of this writing it's 3.9.
 You can use any operating system.
 
-Install development dependencies inside a virtual environment of your choice, for
-example:
+Most development tasks are automated by [nox][nox] which automatically creates virtual
+environments and running the right commands. You'll most likely run the test suite,
+build docs, etc. using nox. Often, you can run `python -m pip install nox` to install
+and use it.
+
+```{tip}
+If you're on Linux or MacOS you might want to use [pipx][pipx] to install nox (and
+optionally pre-commit) to avoid interfering with the system Python installation.
+
+Alternatively install them into the virtual environment you'll be setting up below.
+```
+
+### Running black from source tree
+
+To run the `black` executable from your source tree during development, install _Black_
+locally using editable installation (inside a virtual environment). You can then invoke
+your local source tree _Black_ normally.
+
+**Linux / MacOS**
 
 ```console
 $ python3 -m venv .venv
 $ source .venv/bin/activate
-(.venv)$ pip install -r test_requirements.txt
-(.venv)$ pip install -e .[d]
-(.venv)$ pre-commit install
+(.venv)$ pip install -e .[d,jupyter]
 ```
 
-Before submitting pull requests, run lints and tests with the following commands from
-the root of the black repo:
+**Windows**
 
 ```console
-# Linting
-(.venv)$ pre-commit run -a
+$ py -m venv .venv
+$ .venv\Scripts\activate
+(.venv)$ python -m pip install -e .[d,jupyter]
+```
+
+### Running linters
 
-# Unit tests
-(.venv)$ tox -e py
+_Black_ uses [pre-commit][pre-commit] to manage linting. It has been configured with a
+list of _pre-commit hooks_ that each check a certain aspect of the codebase.
 
-# Optional Fuzz testing
-(.venv)$ tox -e fuzz
+```console
+$ nox -s lint
 ```
 
-### News / Changelog Requirement
+You can also setup pre-commit to automatically run before committing. First, install
+pre-commit similarly to nox and install the Git pre commit hook:
 
-`Black` has CI that will check for an entry corresponding to your PR in `CHANGES.md`. If
-you feel this PR does not require a changelog entry please state that in a comment and a
-maintainer can add a `skip news` label to make the CI pass. Otherwise, please ensure you
-have a line in the following format:
+```console
+$ python -m pip install pre-commit
+$ pre-commit install
+```
 
-```md
-- `Black` is now more awesome (#X)
+### Running tests
+
+Before opening a pull request that involves code, please run the test suite to verify
```

Comment:
I'm actually not sure we should recommend this. Local tests are often annoying and finicky to set up, so why not just rely on our CI?